File: week4/4.3/HighestValueInArray/HighestValueInArray/ViewController.m
//
//  ViewController.m
//  HighestValueInArray
//
//  Created by Eric Lanza on 7/24/18.
//  Copyright © 2018 ETLanza. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    NSArray *array = @[@10000, @9, @77777, @789, @678, @6];
    
    NSNumber *max = [array valueForKeyPath:@"@max.self"];
    NSLog(@"%@", max);
    
    [self highestValueInArray:array];

}

- (void)highestValueInArray:(NSArray *)array
{
    NSArray *sortedArray = [array sortedArrayUsingComparator:^NSComparisonResult(id  _Nonnull obj1, id  _Nonnull obj2)
    {
        return obj1 < obj2;
    }];
    
    NSLog(@"%@", sortedArray);
    NSLog(@"Highest value: %@", sortedArray[0]);
    NSLog(@"Second highest value: %@", sortedArray[1]);
}


@end

File: week6/6.5/PopUpMarkM/PopUpMarkM/ViewController.swift
//
//  ViewController.swift
//  PopUpMarkM
//
//  Created by Eric Lanza on 8/10/18.
//  Copyright © 2018 ETLanza. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    
    @IBOutlet weak var timeDateLabel: UILabel!
    
    // MARK: - Navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        let formatter = DateFormatter()
        
        if segue.identifier == "datePickerSegue" {
            weak var destinationVC = segue.destination as? PopUpViewController
            destinationVC?.showTimePicker = false
            formatter.dateStyle = .medium
            destinationVC?.onSave = { //[weak destinationVC] in
                self.timeDateLabel.text = formatter.string(from: destinationVC!.datePicker.date)
            }
        }
        
        if segue.identifier == "timePickerSegue" {
            let destinationVC = segue.destination as! PopUpViewController
            destinationVC.showTimePicker = true
            formatter.timeStyle = .medium
            destinationVC.onSave = { [weak self, weak destinationVC] in
                self?.timeDateLabel.text = formatter.string(from: destinationVC!.datePicker.date)
            }
        }
    }
}
